src/crypto/keys.ts

```diff
@@ -0,0 +1,30 @@
+import { SecretKey } from 'umbral-pre';
+
+import { toBytes } from '../utils';
+
+import { keccakDigest } from './api';
+import { UMBRAL_KEYING_MATERIAL_BYTES_LENGTH } from './constants';
+
+export class UmbralKeyingMaterial {
+  private readonly keyingMaterial: Uint8Array;
+
+  constructor(keyingMaterial: Uint8Array) {
+    if (keyingMaterial.length !== UMBRAL_KEYING_MATERIAL_BYTES_LENGTH) {
+      throw Error(
+        `Expected keyingMaterial to be ${UMBRAL_KEYING_MATERIAL_BYTES_LENGTH} bytes long, received ${keyingMaterial.length} bytes instead`
+      );
+    }
+    this.keyingMaterial = keyingMaterial;
+  }
+
+  public deriveSecretKeyFromLabel(label: string): SecretKey {
```

Comment:
Hm, why can't you use `SecretKeyFactory` here?

Reply:
No reason, I just forgot about `SecretKeyFactory::makeKey`